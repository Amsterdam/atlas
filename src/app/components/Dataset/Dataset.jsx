import React from 'react'
import { AngularWrapper } from 'react-angular'
import classNames from 'classnames'
import angular from 'angular'
import PropTypes from 'prop-types'
import DATA_SELECTION_CONFIG from '../../../shared/services/data-selection/data-selection-config'
import DatasetActiveFilters from '../../containers/DatasetActiveFiltersContainer'
import { DEFAULT_DATASET } from '../../../shared/ducks/datasets/data/data'
import MaxPageMessage from '../PanelMessages/MaxPageMessage'
import NoResultsForSearchType from '../Messages/NoResultsForSearchType'
import Catalog from './Catalog/Catalog'
import ShareBar from '../ShareBar/ShareBar'
import '../../angularModules'

const Dataset = ({
  setPage,
  activeFilters,
  results: { numberOfRecords, numberOfPages, data, filters: availableFilters },
  page: currentPage,
  apiSpecification,
}) => {
  const showFilters = numberOfRecords > 0
  const { MAX_AVAILABLE_PAGES } = DATA_SELECTION_CONFIG.datasets[DEFAULT_DATASET]
  const showMessageMaxPages = MAX_AVAILABLE_PAGES && currentPage > MAX_AVAILABLE_PAGES

  const widthClass = classNames({
    'u-col-sm--12': !showFilters,
    'u-col-sm--9': showFilters,
  })

  if (numberOfRecords === 0) {
    return (
      <NoResultsForSearchType
        message={`Tip: maak de zoekcriteria minder specifiek. Of probeer in plaats van zoeken eens
        de optie 'Alle datasets tonen' en filter vervolgens op thema.`}
        hideLoginLink
      />
    )
  }

  return (
    <div>
      <DatasetActiveFilters />

      <div className="u-grid qa-data-grid">
        <div className="u-row">
          {showFilters && (
            <div className="u-col-sm--3 c-data-selection__available-filters">
              <AngularWrapper
                moduleName="dpDataSelectionAvailableFiltersWrapper"
                component="dpDataSelectionAvailableFilters"
                dependencies={['atlas']}
                angularInstance={angular}
                bindings={{
                  availableFilters,
                  activeFilters,
                }}
                interpolateBindings={{
                  dataset: DEFAULT_DATASET,
                }}
              />
            </div>
          )}
          <div className={widthClass}>
            {showMessageMaxPages && <MaxPageMessage maxAvailablePages={MAX_AVAILABLE_PAGES} />}

            <div>
              <Catalog content={data} catalogFilters={apiSpecification} />
            </div>
            <AngularWrapper
              moduleName="dpDataSelectionPaginationWrapper"
              component="dpDataSelectionPagination"
              dependencies={['atlas']}
              angularInstance={angular}
              bindings={{
                currentPage,
                numberOfPages,
                setPage,
              }}
            />
            <div className="u-row">
              <div className="u-col-sm--12">
                <div className="u-margin__top--4">
                  <ShareBar />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

Dataset.propTypes = {
  activeFilters: PropTypes.shape({}).isRequired,
  page: PropTypes.number.isRequired,
  setPage: PropTypes.func.isRequired,
  apiSpecification: PropTypes.shape().isRequired,
  results: PropTypes.shape({
    numberOfRecords: PropTypes.number,
    numberOfPages: PropTypes.number,
    filters: PropTypes.arrayOf(PropTypes.object),
    data: PropTypes.arrayOf(PropTypes.object),
  }).isRequired,
}

export default Dataset
