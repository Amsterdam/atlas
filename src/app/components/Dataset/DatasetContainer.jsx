import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  getApiSpecificationData,
  getPage,
  getResults
} from '../../../shared/ducks/datasets/datasets';
import { getActiveFilters } from '../../../shared/ducks/filters/filters';
import { setPage as setPageActions } from '../../../shared/ducks/datasets/data/data';
import Dataset from './Dataset';

const mapStateToProps = (state) => ({
  page: getPage(state),
  activeFilters: getActiveFilters(state),
  results: getResults(state),
  apiSpecification: getApiSpecificationData(state)
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  setPage: setPageActions
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Dataset);
