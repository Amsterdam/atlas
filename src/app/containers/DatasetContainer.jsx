import connect from 'react-redux/es/connect/connect';
import { bindActionCreators } from 'redux';
import {
  dataIsLoading,
  getAuthError,
  getDatasetApiSpecification,
  getPage,
  getResults
} from '../../shared/ducks/datasets/datasets';
import { getActiveFilters } from '../../shared/ducks/filters/filters';
import { getUser } from '../../shared/ducks/user/user';
import { setPage as setPageAction } from '../../shared/ducks/datasets/data/data';
import Dataset from '../components/Dataset/Dataset';

const mapStateToProps = (state) => ({
  isLoading: dataIsLoading(state),
  authError: getAuthError(state),
  page: getPage(state),
  activeFilters: getActiveFilters(state),
  results: getResults(state),
  user: getUser(state),
  apiSpecification: getDatasetApiSpecification(state)
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  setPage: setPageAction
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Dataset);
