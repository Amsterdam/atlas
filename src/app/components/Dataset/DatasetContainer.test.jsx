import React from 'react'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import DatasetContainer from './DatasetContainer'
import Dataset from './Dataset'

const initialState = {
  datasets: {
    datasetData: {
      page: 1,
      result: {},
    },
    datasetApiSpecification: {
      data: {},
    },
  },
  filter: {
    filters: {},
  },
}

describe('Dataset', () => {
  it('should render the container', () => {
    const store = configureMockStore()({ ...initialState })
    const component = shallow(<DatasetContainer />, { context: { store } })
    expect(component).toMatchSnapshot()
  })

  it('should render the component', () => {
    const component = shallow(
      <Dataset
        activeFilters={{}}
        apiSpecification={{}}
        page={1}
        results={{ data: [] }}
        setPage={jest.fn}
      />,
    )
    expect(component).toMatchSnapshot()
  })
})
