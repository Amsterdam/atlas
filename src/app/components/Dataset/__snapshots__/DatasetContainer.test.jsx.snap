// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dataset should render the component 1`] = `
<div>
  <Connect(DatasetActiveFilters) />
  <div
    className="u-grid qa-data-grid"
  >
    <div
      className="u-row"
    >
      <div
        className="u-col-sm--12"
      >
        <div>
          <Catalog
            catalogFilters={Object {}}
            content={Array []}
          />
        </div>
        <AngularWrapper
          bindings={
            Object {
              "currentPage": 1,
              "numberOfPages": undefined,
              "setPage": [Function],
            }
          }
          component="dpDataSelectionPagination"
          dependencies={
            Array [
              "atlas",
            ]
          }
          interpolateBindings={Object {}}
          moduleName="dpDataSelectionPaginationWrapper"
        />
        <div
          className="u-row"
        >
          <div
            className="u-col-sm--12"
          >
            <div
              className="u-margin__top--4"
            >
              <Connect(ShareBar) />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dataset should render the container 1`] = `
<Dataset
  activeFilters={Object {}}
  apiSpecification={Object {}}
  page={1}
  results={Object {}}
  setPage={[Function]}
/>
`;
